node/src/Pos/Block/Network/Retrieval.hs

```diff
@@ -98,6 +99,7 @@ retrievalWorkerImpl SendActions {..} =
                     pure (handleBlockRetrievalFromQueue nodeId task)
                 (_, Just (nodeId, rHeader))  ->
                     pure (handleHeadersRecovery nodeId rHeader)
+        startTimer keepAliveTimer
```

Comment:
What's this for?

Reply:
Oh, this will be the same timer that the subscription worker uses, and this `startTimer` call will restart it so that the subscription won't send the keepalive for another 30 seconds?

Reply:
If so I think it deserves a comment.

Reply:
Yes.

Reply:
Right, I thought that it might be a good idea to explain this anyway. Pushed appropriate commit.